src/aggregation/staging_collector.rs

```diff
@@ -0,0 +1,83 @@
+use super::agg_req_with_accessor::AggregationsWithAccessor;
+use super::intermediate_agg_result::IntermediateAggregationResults;
+use super::segment_agg_result::SegmentAggregationCollector;
+use crate::DocId;
+
+pub(crate) const DOC_BLOCK_SIZE: usize = 64;
```

Comment:
I would have thought 64 was a tad small. Have you tried different sizes?

Reply:
I did tests but only with the old caching variant, which worked a little different
Eventually we'll probably have different sizes for different use cases